src/vcpkg/commands.setinstalled.cpp

```diff
@@ -36,6 +39,85 @@ namespace vcpkg::Commands::SetInstalled
         nullptr,
     };
 
+    static void create_dependency_graph_json(const ActionPlan& action_plan)
+    {
+        auto gh_ref = get_environment_variable("GITHUB_REF").value_or_exit(VCPKG_LINE_INFO);
+        auto gh_sha = get_environment_variable("GITHUB_SHA").value_or_exit(VCPKG_LINE_INFO);
+        auto gh_owner = get_environment_variable("GITHUB_OWNER").value_or_exit(VCPKG_LINE_INFO);
+        auto gh_repo = get_environment_variable("GITHUB_REPO").value_or_exit(VCPKG_LINE_INFO);
+        auto gh_token = get_environment_variable("GITHUB_TOKEN").value_or_exit(VCPKG_LINE_INFO);
```

Comment:
These should all be initially captured by `VcpkgCmdArguments` and then plumbed into this function.